app/src/test/java/org/oppia/android/app/customview/interaction/FractionInputInteractionViewTest.kt

```diff
@@ -0,0 +1,544 @@
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [FractionInputInteractionView]. */
+@RunWith(AndroidJUnit4::class)
+@Config(
+  application = FractionInputInteractionViewTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+@LooperMode(LooperMode.Mode.PAUSED)
+
+class FractionInputInteractionViewTest {
+
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+
+  @get:Rule
+  val oppiaTestRule = OppiaTestRule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var profileTestHelper: ProfileTestHelper
+
+  @Inject
+  lateinit var context: Context
+
+  @Inject
+  lateinit var editTextInputAction: EditTextInputAction
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Test
+  fun testFractionInput_withNoInput_hasCorrectPendingAnswerType() {
+    val activityScenario = ActivityScenario.launch(
+      InputInteractionViewTestActivity::class.java
+    )
+    activityScenario.onActivity { activity ->
+      val pendingAnswer = activity.fractionInteractionViewModel.getPendingAnswer()
+      Truth.assertThat(pendingAnswer.answer).isInstanceOf(InteractionObject::class.java)
+      Truth.assertThat(pendingAnswer.answer.fraction.denominator).isEqualTo(0)
+      Truth.assertThat(pendingAnswer.answer.fraction.numerator).isEqualTo(0)
+      Truth.assertThat(pendingAnswer.answer.fraction.wholeNumber).isEqualTo(0)
+    }
+  }
+
+  @Test
+  @DisableAccessibilityChecks
+  fun testFractionInput_withNegativeNumber_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(
+      InputInteractionViewTestActivity::class.java
+    )
+    Espresso.onView(ViewMatchers.withId(R.id.test_fraction_input_interaction_view))
+      .perform(editTextInputAction.appendText("-9"))
+    activityScenario.onActivity { activity ->
+      val pendingAnswer = activity.fractionInteractionViewModel.getPendingAnswer()
+      Truth.assertThat(pendingAnswer.answer).isInstanceOf(InteractionObject::class.java)
+      Truth.assertThat(pendingAnswer.answer.objectTypeCase).isEqualTo(
+        InteractionObject.ObjectTypeCase.FRACTION
+      )
+      Truth.assertThat(pendingAnswer.answer.fraction.isNegative).isEqualTo(true)
+      Truth.assertThat(pendingAnswer.answer.fraction.wholeNumber).isEqualTo(9)
+    }
+  }
+
+  @Test
+  @DisableAccessibilityChecks
+  fun testFractionInput_withWholeNumber_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(
+      InputInteractionViewTestActivity::class.java
```

Comment:
Replace with the new Fraction specific test activity.

Reply:
> Replace with the new Fraction specific test activity.

I tried the way you suggested, by creating FractionInputInteractionViewTestActivity to replace InputInteractionViewTestActivity but now  tests are failing in FractionInputInteractionViewTest.kt file.

![Screenshot from 2022-03-19 00-30-43](https://user-images.githubusercontent.com/78796264/159068295-f9545ee9-bd7a-448a-8914-fd0d45745df6.png)

Specific error:--
'java.lang.RuntimeException: Unable to resolve activity for Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=org.oppia.android/.app.testing.FractionInputInteractionViewTestActivity } -- see https://github.com/robolectric/robolectric/pull/4736 for details'

Please do suggest how should I proceed?

Reply:
Could you share your `FractionInputInteractionViewTestActivity`?